src/test/java/com/ably/kafka/connect/DefaultAblyBatchClientTest.java

```diff
@@ -0,0 +1,68 @@
+package com.ably.kafka.connect;
+
+import com.ably.kafka.connect.client.DefaultAblyBatchClient;
+import com.ably.kafka.connect.config.ChannelConfig;
+import com.ably.kafka.connect.config.ChannelSinkConnectorConfig;
+import com.ably.kafka.connect.config.ConfigValueEvaluator;
+import com.ably.kafka.connect.config.DefaultChannelConfig;
+import com.ably.kafka.connect.mapping.ChannelSinkMapping;
+import com.ably.kafka.connect.mapping.DefaultChannelSinkMapping;
+import com.ably.kafka.connect.mapping.DefaultMessageSinkMapping;
+import com.ably.kafka.connect.mapping.MessageSinkMapping;
+import io.ably.lib.types.AblyException;
+import io.ably.lib.types.Message;
+import org.apache.kafka.connect.data.Schema;
+import org.apache.kafka.connect.sink.SinkRecord;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import org.junit.jupiter.api.Test;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+
+public class DefaultAblyBatchClientTest {
+
+    @Test
+    public void testGroupMessagesByChannel() throws AblyException {
+
+        final String STATIC_CHANNEL_NAME = "channel_#{topic}";
+        final ChannelSinkConnectorConfig connectorConfig = new ChannelSinkConnectorConfig(Map.of("channel",
+                STATIC_CHANNEL_NAME, "client.key", "test-key", "client.id", "test-id"));
+        final ConfigValueEvaluator configValueEvaluator = new ConfigValueEvaluator();
+        final ChannelConfig channelConfig = new DefaultChannelConfig(connectorConfig);
+        final ChannelSinkMapping channelSinkMapping = new DefaultChannelSinkMapping(configValueEvaluator, channelConfig);
+        final MessageSinkMapping messageSinkMapping = new DefaultMessageSinkMapping(connectorConfig, configValueEvaluator);
+        DefaultAblyBatchClient client = new DefaultAblyBatchClient(connectorConfig, channelSinkMapping,
+                messageSinkMapping, configValueEvaluator);
+
+        List<SinkRecord> sinkRecords = new ArrayList<>();
```

Comment:
Can we assume Java 11 and make use of the newer `List.of()` convenience methods here? Says here that Java 8 support is deprecated in Confluence platform: https://docs.confluent.io/platform/current/installation/versions-interoperability.html#java

Are we supporting Java 8 intentionally?

Reply:
changed to use ListOf(